src/Metrics/Transports/OpenTelemetryMetricTransport.php

```diff
@@ -7,98 +7,100 @@
 use OpenTelemetry\API\Metrics\MeterInterface;
 use OpenTelemetry\API\Metrics\ObserverInterface;
 use OpenTelemetry\Context\ContextInterface;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\Metric;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\Type;
 use Shopware\Core\Framework\Telemetry\TelemetryException;
-use Shopware\Core\Framework\Telemetry\Metrics\Metric\Counter;
-use Shopware\Core\Framework\Telemetry\Metrics\Metric\Gauge;
-use Shopware\Core\Framework\Telemetry\Metrics\Metric\Histogram;
-use Shopware\Core\Framework\Telemetry\Metrics\Metric\MetricInterface;
-use Shopware\Core\Framework\Telemetry\Metrics\Metric\UpDownCounter;
 use Shopware\Core\Framework\Telemetry\Metrics\MetricTransportInterface;
+use Shopware\OpenTelemetry\Metrics\MetricNameFormatter;
 
 /**
  * @phpstan-type Attributes iterable<non-empty-string, string|bool|float|int>
  */
-class OpenTelemetryMetricTransport implements MetricTransportInterface
+readonly class OpenTelemetryMetricTransport implements MetricTransportInterface
 {
     public function __construct(
-        private string $namespace,
-        private MeterInterface $meter,
+        private MetricNameFormatter $formatter,
+        private MeterInterface      $meter,
     ) {}
 
-    public function emit(MetricInterface $metric): void
+    public function emit(Metric $metric): void
     {
-        $attributes = [];
+        $attributes = $metric->labels;
         $context = null;
 
-        match (true) {
-            $metric instanceof UpDownCounter => $this->handleUpDownCounter($metric, $attributes, $context),
-            $metric instanceof Counter => $this->handleCounter($metric, $attributes, $context),
-            $metric instanceof Histogram => $this->handleHistogram($metric, $attributes, $context),
-            $metric instanceof Gauge => $this->handleGauge($metric, $attributes, $context),
+        match ($metric->type) {
+            Type::UPDOWN_COUNTER => $this->handleUpDownCounter($metric, $attributes, $context),
+            Type::COUNTER => $this->handleCounter($metric, $attributes, $context),
+            Type::HISTOGRAM => $this->handleHistogram($metric, $attributes, $context),
+            Type::GAUGE => $this->handleGauge($metric, $attributes, $context),
             default => throw TelemetryException::metricNotSupported($metric, $this),
```

Comment:
Good point. But as far as  I remember, if `ImmediateStalenessHandlerFactory` is used
`$meter->createHistogram(...)->record(...)` would not work, it would have to be written as two-liner
```
$histogram = $meter->createHistogram(...);
$histogram->record(...);
```
It was one of the reasons why I kept separate methods.